src/Capgemini.PowerApps.PackageDeployerTemplate/PackageTemplateBase.cs

```diff
@@ -22,193 +17,77 @@ public abstract class PackageTemplateBase : ImportExtension
-                this.PackageLog.LogExecuteMultipleErrors(executeMultipleResponse);
-            }
+            this.BeforeAnything();
         }
 
-        private void ActivateProcesses(IEnumerable<string> processesToActivate)
+        public void BeforeAnything()
         {
-            if (processesToActivate is null || !processesToActivate.Any())
-            {
-                this.PackageLog.Log("No processes to activate have been configured.");
-                return;
-            }
+            this.PackageLog.Log($"{nameof(PackageTemplateBase)}.{nameof(BeforeAnything)} running...", TraceEventType.Information);
 
-            var executeMultipleResponse = this.CrmSvc.SetRecordStateByAttribute("workflow", 1, 2, "name", processesToActivate);
-            if (executeMultipleResponse.IsFaulted)
+            if (this.ConfigDataStorage.ActivateDeactivateSLAs)
             {
-                this.PackageLog.Log($"Error activating processes.", TraceEventType.Error);
-                this.PackageLog.LogExecuteMultipleErrors(executeMultipleResponse);
+                this.SlaActivatorService.Deactivate();
             }
-        }
+            this.DataImporterService.Import(this.ConfigDataStorage.DataImports?.Where(c => c.ImportBeforeSolutions), this.PackageFolderPath);
 
-        private void DeactivateSlas()
-        {
-            var executeMultipleResponse = this.CrmSvc.SetRecordStateByAttribute("sla", 0, 1, "statecode", new object[] { 1 });
-            if (executeMultipleResponse.IsFaulted)
-            {
-                this.PackageLog.Log($"Error deactivating SLAs.", TraceEventType.Error);
-                this.PackageLog.LogExecuteMultipleErrors(executeMultipleResponse);
-            }
+            this.PackageLog.Log($"{nameof(PackageTemplateBase)}.{nameof(BeforeAnything)} completed.", TraceEventType.Information);
         }
 
-        public override void InitializeCustomExtension()
+        public override bool BeforeImportStage()
         {
-            this.PackageLog.Log($"Initializing {nameof(PackageTemplateBase)} extension.");
-
-            this.ConfigDataStorage = ConfigDataStorage.Load(this.ImportConfigFilePath);
-
-            this.DataImporter = new DataImporter(
-                this.PackageLog,
-                new EntityRepository(
-                    (IOrganizationService)this.CrmSvc.OrganizationWebProxyClient ?? CrmSvc.OrganizationServiceProxy,
-                    new ServiceRetryExecutor()));
-
-            // Previously in the BeforeImportStage method but this does not run before solution import.
-            if(this.ConfigDataStorage.ActivateDeactivateSLAs)
-            {
-                this.DeactivateSlas();
-            }
-            this.ImportData(this.ConfigDataStorage.DataImports?.Where(c => c.ImportBeforeSolutions));
+            this.PackageLog.Log($"{nameof(PackageTemplateBase)}.{nameof(BeforeImportStage)} running...", TraceEventType.Information);
```

Comment:
This is going to result in funny behaviour when overriding. For example - 

```csharp
public override bool BeforeImportStage() 
{
    base.BeforeImportStage();

    // Some additional stuff here
}
```

Will log running and completed before doing additional work.

Reply:
It's worth checking if these are needed as the Package Deployer itself has quite good logging

Reply:
I'll double-check the logs (which I agree have been really good) but I wanted it to signify when our custom code was running. I'll explore some more.

Reply:
Below is an extract of the log ran locally:
```
PackageDeployer Information: 8 : Message: ************** BeforeImportStage *****************
PackageDeployer Information: 8 : Message: User Code Execution : OP=BeforeImportStage : Status=Execute
PackageDeployer Information: 8 : Message: PackageTemplateBase.BeforeImportStage running...
PackageDeployer Information: 8 : Message: PackageTemplateBase.BeforeImportStage completed.
PackageDeployer Information: 8 : Message: User Code Execution : OP=BeforeImportStage : Status=Complete : Duration=00:00:00.0036674
PackageDeployer Information: 8 : Message: Pre-import process complete.--Complete
```

I agree that these statements are pointless while there isn't anything beginning executed but I think they work well to mark when our base code is executing compared with the user's own code. 

Still happy to remove if you think we should 😄

Reply:
Looking at the logs, that makes more sense now. I was just thinking it would say `BeforeImportStage` was completed and then we'd have some additional code, but as it's wrapped in the Package Deployer's start/end logs for that stage it won't give the wrong impression.